dhall-toml/src/Dhall/Toml.hs

```diff
@@ -43,13 +57,20 @@ data CompileError
     --   implemented it yet
     --   NOTE: the only way to get this error is through unions
     | HeterogeneousArray (Expr Void Void)
+    | InvalidToml TomlParseError
+    | InternalError String
+    | MissingKey String
```

Comment:
I think it would be better to have separate error types for each conversion direction.